### PR TITLE
Use role based creds

### PR DESCRIPTION
**Pull request by russellpierce**
Switch over to use role based access to S3 buckets.  Passes existing tests.  To use the role successfully note https://github.com/zapier/data-science-env/commit/19a7c98f558ad299f80fb9f4cf6013dbeec8fb18

### CODE REVIEW

**Comment by russellpierce** [comment]
Heya @shalgrim , this change needs to be on worker boxes before they can live without AWS keys.  There probably are some other edges where we use those creds... but this was one of the obvious ones.

File: R/internal.R [hunk 1]
```diff
@@ -1,3 +1,6 @@
+#' @importFrom reticulate import
+boto <- reticulate::import("boto", delay_load = TRUE)
```
Comment (shalgrim) [inline R/internal.R:2]: does reticulate not support boto3?

File: R/internal.R [hunk 2]
```diff
@@ -88,11 +97,11 @@ uploadToS3 <- function(data, bucket, split_files) {
 
 #' @importFrom "aws.s3" "delete_object"
 #' @importFrom parallel mclapply
-deletePrefix <- function(prefix, bucket, split_files){
-  parallel::mclapply(1:split_files, function(i) {
-    s3Name = paste(prefix, ".", formatC(i, width = 4, format = "d", flag = "0"), ".psv.gz", sep="")
-    delete_object(s3Name, bucket)
-  })
+deletePrefix <- function(prefix, bucket, split_files) {
```
Comment (shalgrim) [inline R/internal.R:101]: This looks like it's only going to delete keys of a certain format within the prefix/folder, is that right? And we have to know the format?

**Review by shalgrim — APPROVED** [review]

**Comment by russellpierce** [comment]
[Update package:redshiftTools to take advantage of role based access to S3](https://trello.com/c/FgXrPZSA/430-update-packageredshifttools-to-take-advantage-of-role-based-access-to-s3)

**Comment by shalgrim** [comment]
will re-open as merge from develop

**Pull request closed without merging**